examples/command_bot.rs

```diff
@@ -0,0 +1,123 @@
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::{env, process::exit};
+
+use js_int::UInt;
+use matrix_sdk::{
+    self,
+    events::room::message::{MessageEvent, MessageEventContent, TextMessageEventContent},
+    AsyncClient, AsyncClientConfig, EventEmitter, Room, SyncSettings,
+};
+use tokio::sync::{Mutex, RwLock};
+use url::Url;
+
+struct CommandBot {
+    /// This clone of the `AsyncClient` will send requests to the server,
+    /// while the other keeps us in sync with the server using `sync_forever`.
+    client: Mutex<AsyncClient>,
```

Comment:
So this is behind a mutex probably because `room_send()` takes `&mut self` but I think that's a mistake or a remnant from the time when the base client wasn't behind a lock/mutex.

I think we can change all the methods that do API requests to take `&self` instead of `&mut self` and this will in turn make this example simpler.

If you want you can change this in this PR or we can leave it for some other time.

Reply:
Cool yeah, I was wondering about that while I was trying to figure out the deadlock, I will make the changes.